From 4a4794e041edf85cb67b9fbba66ad1a22fdcaaec Mon Sep 17 00:00:00 2001
From: Eric Bavier <bavier@posteo.net>
Date: Sat, 29 Jan 2022 00:10:20 -0600
Subject: [PATCH] Fixes linux build with glibc-2.32.

* cpulimit.c: Guard inclusion of <sys/sysctl.h> which is needed only for
  __APPLE__ and is deprecated and unavailable starting with glibc-2.32.
---
 src/cpulimit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Submitted upstream at https://github.com/opsengine/cpulimit/pull/105

diff --git a/src/cpulimit.c b/src/cpulimit.c
index 50eabea..be8cf22 100644
--- a/src/cpulimit.c
+++ b/src/cpulimit.c
@@ -38,13 +38,13 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/time.h>
-#include <sys/sysctl.h>
 #include <sys/resource.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 
 #ifdef __APPLE__ || __FREEBSD__
 #include <libgen.h>
+#include <sys/sysctl.h>
 #endif
 
 #include "process_group.h"
-- 
2.34.0

